src/sources/prometheus/parser.rs

```diff
@@ -412,63 +119,37 @@ pub fn parse(packet: &str) -> Result<Vec<Metric>, ParserError> {
                     result.push(hist);
                 }
             }
-            ParserType::Summary => {
+            GroupKind::Summary(vec) => {
                 let mut aggregates = IndexMap::new();
 
-                for line in group {
-                    let metric = parse_metric(&line)?;
-                    let mut tags = metric.tags;
-                    let bucket = tags.remove("quantile");
-
-                    let (name, suffix) =
-                        if metric.name.ends_with("_sum") || metric.name.ends_with("_count") {
-                            let v: Vec<_> = metric.name.rsplitn(2, '_').collect();
-                            (v[1], v[0])
-                        } else {
-                            (&metric.name[..], "")
-                        };
-
-                    let mut id: Vec<_> = tags.iter().collect();
-                    id.sort();
-                    let group_key = format!("{:?}", id);
-
-                    let aggregate = aggregates.entry(group_key).or_insert(ParserAggregate {
-                        name: name.to_owned(),
+                for metric in vec {
+                    let tags = metric.get_labels().clone();
+                    let aggregate = aggregates.entry(tags.clone()).or_insert(ParserAggregate {
```

Comment:
Do all lines for a given tags have to be consecutive? If yes, we can get rig of `IndexMap` and just use a `Vec` (like in `fn group_metrics`).